tools/dockerise/__init__.py

```diff
@@ -106,24 +122,33 @@ def get_selected_platform():
             ).communicate()[0],
         )
 
-        names = [
-            tag.split(":")[-1]
-            for tag in img_info[0]["RepoTags"]
-            if tag != "{}:selected".format(repository) and tag.startswith("{}:".format(repository))
-        ]
-        if len(names) == 0:
-            print("ERROR the currently selected platform is a dangling tag.")
-            print("      The system is unable to work out what platform this was and will need to be reset")
-            print("      run `./b target {platform}` to correct this")
-            exit(1)
-        elif len(names) == 1:
-            return names[0]
+        if len(img_info) > 0:
+            names = [
+                tag.split(":")[-1]
+                for tag in img_info[0]["RepoTags"]
+                if tag != "{}:selected".format(repository) and tag.startswith("{}:".format(repository))
```

Comment:
I think this is a little more readable
```suggestion
                if tag != f"{repository}:selected" and tag.startswith(f"{repository}:")
```